opty/direct_collocation.py

```diff
@@ -686,6 +686,35 @@ def parse_free(self, free):
 
         return parse_free(free, n, q, N, variable_duration)
 
+    def time_vector(self, t0=0.0, solution=None):
+        """Returns the time instances of the problem as an numpy ndarray.
+
+        Parameters
+        ----------
+        t0 : float, optional
+            The initial time of the problem. Default is 0.0.
+        solution : (n*N + q*N + r + s)-ndarray, optional
+            The solution to to problem. Needed if the interval is variable.
+
+        Returns
+        -------
+        A numpy num_collocation_nodes-array of time instances.
+
+        """
+        if isinstance(self.collocator.node_time_interval, sm.Symbol):
```

Comment:
There is a flag `self.collocator._variable_duration` that should be used for this check.